engine/crates/composition/src/ingest_subgraph/directives.rs

```diff
@@ -88,6 +91,60 @@ pub(super) fn ingest_directives(
             }
         }
 
+        if directive_matcher.is_authenticated(directive_name) {
+            subgraphs.insert_authenticated(directives);
+        }
+
+        if directive_matcher.is_requires_scope(directive_name) {
+            let scopes = directive
+                .node
+                .get_argument("scopes")
+                .into_iter()
+                .flat_map(|scopes| match &scopes.node {
+                    ConstValue::List(list) => Some(list),
+                    _ => None,
+                })
+                .flatten();
+            for scope in scopes {
+                let inner_scopes: Vec<subgraphs::StringId> = match scope {
+                    ConstValue::List(scopes) => scopes
+                        .iter()
+                        .filter_map(|scope| match scope {
+                            ConstValue::String(string) => Some(subgraphs.strings.intern(string.as_str())),
+                            _ => None,
+                        })
+                        .collect(),
+                    _ => vec![],
+                };
```

Comment:
This is also not great. We could lean on to the Serialize impl for ConstValue but I did that in another place (from_sdl()) and it's barely less verbose.